python/triton/runtime/jit.py

```diff
@@ -668,7 +668,7 @@ def run(self, *args, grid, warmup, **kwargs):
         kernel = kernel_cache.get(key, None)
 
         # Kernel is not cached; we have to compile.
-        if kernel is None:
+        if kernel is None or knobs.compilation.always_compile:
```

Comment:
Attention place. For me it looks safe and coincides with the nature of this knobs.

I came across this because the cache is reused even if the environment variables are changed, which can be fixed in https://github.com/triton-lang/triton/pull/8342.

Reply:
I don't think this is correct. Always compile just means we don't fetch from the disk.

Reply:
> TRITON_ALWAYS_COMPILE=1 forces to compile kernels regardless of cache hit.

Hm, maybe it's worth change the wording in that case? In this case, a cache hit is occurring, not of files, but of objects in memory, but it seems to fit the wording in the documentation.

Reply:
there are multiple levels of caching, we could change the readme but current implementation is what we want